src/base/bittorrent/session.cpp

```diff
@@ -1444,12 +1444,16 @@ void Session::configureNetworkInterfaces(lt::settings_pack &settingsPack)
         const QHostAddress addr {ip};
         if (!addr.isNull())
         {
-            const QString ip = ((addr.protocol() == QAbstractSocket::IPv6Protocol)
-                          ? ('[' + Utils::Net::canonicalIPv6Addr(addr).toString() + ']')
+            const bool isIPv6 = addr.protocol() == QAbstractSocket::IPv6Protocol;
+            const QString ip = (isIPv6
+                          ? (Utils::Net::canonicalIPv6Addr(addr).toString())
                           : addr.toString());
-            endpoints << (ip + portString);
 
-            if ((ip != "0.0.0.0") && (ip != "[::]"))
+            endpoints << (isIPv6
+                          ? QString('[' + ip + ']') + portString
```

Comment:
Some clarifications:
We need to use `QString` here to 'combine' the three strings into one `QString`, so that we end up with `QString + QString`, because each of the options for the ternary operator need to have the same type. Otherwise compilation will fail.